File: ims-frontend/src/services/zoneService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/zones";

export const getZones = () => axios.get(API_URL);

export const getZonesByBrand = (brandId) =>
  axios.get(`${API_URL}/brand/${brandId}`);

export const createZone = (zone) => axios.post(API_URL, zone);

export const updateZone = (id, zone) =>
  axios.put(`${API_URL}/${id}`, zone);

export const deleteZone = (id) =>
  axios.delete(`${API_URL}/${id}`);

File: ims/src/main/java/com/codeb/ims/service/ZoneService.java
package com.codeb.ims.service;

import com.codeb.ims.model.Zone;
import java.util.List;

public interface ZoneService {

    Zone createZone(Zone zone);

    List<Zone> getAllZones();

    List<Zone> getZonesByBrand(Long brandId);

    Zone updateZone(Long id, Zone zone);

    void deleteZone(Long id);
}